### PR TITLE
fix(backup): customize the timeout of backup binary execution

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#8954 longhorn/longhorn#8319

#### What this PR does / why we need it:

Add a default setting `backupExecutionTimeout`

### CODE REVIEW

File: types/setting.go [hunk 1]
```diff
@@ -409,6 +412,19 @@ var (
 		},
 	}
 
+	SettingDefinitionBackupEngineBinaryTimeout = SettingDefinition{
+		DisplayName: "Backup Engine Binary Timeout",
```
Comment (derekbit) [inline types/setting.go:416]: Can we replace the name with a better one without `Binary`?

Comment (mantissahz) [reply types/setting.go:416]: Sure, any suggestion?

Comment (mantissahz) [reply types/setting.go]: Modified to `BackupExecuteTimeout` for `backup-execute-timeout`. Thanks

Comment (derekbit) [reply types/setting.go]: No for now. My thought is using a generic name.

Comment (mantissahz) [reply types/setting.go]: Modified to `BackupExecutionTimeout`: `backup-execute-timeout`,

File: types/setting.go [hunk 2]
```diff
@@ -409,6 +412,19 @@ var (
 		},
 	}
 
+	SettingDefinitionBackupExecutionTimeout = SettingDefinition{
+		DisplayName: "Backup Execution Timeout",
+		Description: "In minutes. This setting determines the timeout for the backup engine execution. Set to 0 to disable the timeout.",
```
Comment (derekbit) [inline types/setting.go:417]: Suggest rewording the `Set to 0 to disable the timeout.` as I suggested in longhorn/longhorn.

Comment (mantissahz) [reply types/setting.go]: Modified. Thanks.

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
As we discussed, the timeout `0` (infinite wait) is not allowed.

File: types/setting.go [hunk 3]
```diff
@@ -411,6 +414,19 @@ var (
 		},
 	}
 
+	SettingDefinitionBackupExecutionTimeout = SettingDefinition{
+		DisplayName: "Backup Execution Timeout",
+		Description: "In minutes. This setting determines the timeout for the backup engine execution. The default value is 1.",
```
Comment (derekbit) [inline types/setting.go:419]: Waiting for the review of the description.

**Review by derekbit — DISMISSED** [review]
In general, LGTM.

File: types/setting.go (hunk 3 above)
Comment (mantissahz) [reply types/setting.go]: Modified to `Number of minutes that Longhorn allows for backup execution. The default value is "1"`

**Review by derekbit — APPROVED** [review]

**Comment by mantissahz** [comment]
@Mergifyio backport v1.7.x
